R/derive_date_vars.R

```diff
@@ -101,214 +117,730 @@
+#'
+#'   If a component at a higher level than the highest imputation level is
+#'   missing, `NA_character_` is returned. For example, for `highest_imputation
+#'   = "D"` `"2020"` results in `NA_character_` because the month is missing.
+#'
+#'   If `"n"` is specified no imputation is performed, i.e., if any component is
+#'   missing, `NA_character_` is returned.
+#'
+#'   If `"Y"` is specified, `min_dates` or `max_dates` should be specified as
+#'   well. Otherwise, `NA_character_` is returned if the year component is
+#'   missing.
+#'
+#'   *Default*: `"n"`
+#'
+#'   *Permitted Values*: `"Y"` (year, highest level), `"M"` (month), `"D"`
+#'   (day), `"n"` (none, lowest level)
+#'
+#' @param date_imputation The value to impute the day/month when a datepart is
+#'   missing.
+#'
+#'   A character value is expected, either as a
+#'   - format with month and day specified as `"mm-dd"`: e.g. `"06-15"` for the 15th
+#'   of June,
+#'   - or as a keyword: `"first"`, `"mid"`, `"last"` to impute to the first/mid/last
+#'   day/month.
+#'
+#'   Default is `first`.
+#'
+#' @param min_dates Minimum dates
+#'
+#' A list of dates is expected. It is ensured that the imputed date is not
+#' before any of the specified dates, e.g., that the imputed adverse event start
+#' date is not before the first treatment date. Only dates which are in the
+#' range of possible dates of the `dtc` value are considered. The possible dates
+#' are defined by the missing parts of the `dtc` date (see example below). This
+#' ensures that the non-missing parts of the `dtc` date are not changed.
+#' A date or date-time object is expected.
+#' For example
+#'
+#' ```{r echo=TRUE, eval=FALSE}
+#' impute_dtc_dtm(
+#'   "2020-11",
+#'   min_dates = list(
+#'    ymd_hms("2020-12-06T12:12:12"),
+#'    ymd_hms("2020-11-11T11:11:11")
+#'   ),
+#'   highest_imputation = "M"
+#' )
+#' ```
+#'
+#' returns `"2020-11-11T11:11:11"` because the possible dates for `"2020-11"`
+#' range from `"2020-11-01T00:00:00"` to `"2020-11-30T23:59:59"`. Therefore
+#' `"2020-12-06T12:12:12"` is ignored. Returning `"2020-12-06T12:12:12"` would
+#' have changed the month although it is not missing (in the `dtc` date).
+#'
+#' @param max_dates Maximum dates
+#'
+#' A list of dates is expected. It is ensured that the imputed date is not after
+#' any of the specified dates, e.g., that the imputed date is not after the data
+#' cut off date. Only dates which are in the range of possible dates are
+#' considered. A date or date-time object is expected.
+#'
+#' @param preserve Preserve day if month is missing and day is present
+#'
+#' For example `"2019---07"` would return `"2019-06-07` if `preserve = TRUE`
+#' (and `date_imputation = "MID"`).
+#'
+#' Permitted Values: `TRUE`, `FALSE`
+#'
+#' Default: `FALSE`
+#'
+#' @details Usually this computation function can not be used with `%>%`.
+#'
+#' @return A character vector
+#'
+#' @author Samia Kabi, Stefan Bundfuss
+#'
+#' @keywords computation timing
```

Comment:
need to change this to match current recommended keywords and add a family

Reply:
Updated